src/pages/jj/Main/MainComponentMap/MainComponent.js

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react';
+import CommentList from './CommentListMap/CommentListMap';
+
+import uuid from 'react-uuid';
+
+class MainComponent extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      list: [],
+      comment: '',
+      heartColor: false,
+      className: [],
+      bigHeart: false,
+    };
+  }
+
+  // #부모요소에서 받아온 APIKEY값의 comment값
+  componentDidMount() {
+    this.setState({ list: this.props.feedList.comment });
+  }
+
+  // form submit시 함수 실행(댓글 생성)
+  handleSubmit = e => {
+    const { list, className, comment } = this.state;
+    e.preventDefault();
+
+    if (this.state.comment) {
+      const newList = {
+        id: uuid(),
+        comment: comment,
+        class: false,
+      };
+      this.setState({
+        list: [...list, newList],
+        className: [...className, newList.class],
+      });
+      this.setState({ comment: '' });
+    }
+  };
+
+  // 각 댓글에 좋아요 버튼 활성화 함수
+  changeHeartColor = (elId, index) => {
+    const { className } = this.state;
+    let copy = [...className];
+    copy[index] = !copy[index];
+    this.setState({ className: copy });
+  };
```

Comment:
```suggestion
  changeHeartColor = (elId, index) => {
    const { className } = this.state;
    
    this.setState({ className: className.map((class, idx) => idx === index ? !copy[index] : class)});
  };
```
리액트에서는 상태를 업데이트 할 때 불변성을 지켜줘야 합니다. 그렇기 때문에 일반적으로 새로운 복사본을 빈번히 사용하게 되는데요. 다음 문서를 읽으시면서 [불변성](!copy[index])에 대해서도 한번 학습해주세요!